src/common/types/row/tuple_data_collection.cpp

```diff
@@ -169,6 +169,8 @@ void TupleDataCollection::InitializeChunkState(TupleDataChunkState &chunk_state,
 	}
 	InitializeVectorFormat(chunk_state.vector_data, types);
 
+	chunk_state.cached_cast_vectors.clear();
```

Comment:
Unrelated to this PR - but a set of vectors and vector caches sounds suspiciously like a `DataChunk` :)